### PR TITLE
Add setting name on ConfigurationPolicy

### PR DESCRIPTION
**Pull request by yiraeChristineKim**
When creating a policy with multiple manifests and `consolidateManfifest: false` the generated ConfigurationPolicies are names after the Policy with an index appended.

This causes issues when attempting to use extraDependencies such as below.  Adding an additional manifest will change the index causing the existing dependencies to be incorrect.

Allowing setting the name would also make it easier to identify in the UI when reviewing a policy.

Ref: https://issues.redhat.com/browse/ACM-12563

### CODE REVIEW

File: internal/utils.go
Comment (dhaiducek) [inline internal/utils.go:208]: I think we still need to detect whether a name is provided alongside the `object-templates-raw` manifest and add it here.

File: internal/utils.go
Comment (dhaiducek) [inline internal/utils.go:269]: (Optional) I might prefer we pass the parent object to the function and set `policyConf.Manifests[i].Name` in `ApplyDefaults()`: 
```suggestion
					&policyConf.Manifests[i],
```
Then in the conditional above, we detect whether `policyConf.name == Manifests[i].Name` to determine whether it's been overridden, and then set `Manifests[i].Name` directly with the appropriate index if necessary and use that value directly for the name in `buildPolicyTemplate()`. This would also properly handle the case where the same name is given for the manifest and the policy.

**Review by dhaiducek — CHANGES_REQUESTED** [review]
Thanks for the update! I have some comments for consideration.

File: internal/utils.go
Comment (yiraeChristineKim) [reply internal/utils.go:208]: `object-templates-raw` is isPolicyTypeManifest = true so it should have `name` otherwise it throws an error  at `isPolicyTypeManifest, isOcmPolicy, err := isPolicyTypeManifest(` line 188. Do we need to add the name there?

File: internal/utils.go
Comment (yiraeChristineKim) [reply internal/utils.go:269]: I need a` j` (When it has multiple manifest in the Path file) to attach numbers to configurationpolicy name

Comment (dhaiducek) [reply internal/utils.go:269]: Right, I'm proposing the conditional above these lines would use `j` and set the generator struct directly before passing it to `buildPolicyTemplate()`, something like this to detect whether the name was overridden:
```
if j != 0 && policyConf.name != policyConf.Manifests[i].Name {
	policyConf.Manifests[i].Name = fmt.Sprintf("%s%d", policyConf.Manifests[i].Name, j+1)
}
```

File: internal/utils.go
Comment (yiraeChristineKim) [reply internal/utils.go]: configurationPolicy name?

File: internal/utils.go
Comment (yiraeChristineKim) [reply internal/utils.go]: I refactored. I added this func
```
func getConfigurationPolicyName(policyConf *types.PolicyConfig, manifestGroupIndex, policyNum,
	manifestIndex int, useManifestName bool,
) string {
	// Do not append a number to configName when it is used for the first time.
	var configName string

	if useManifestName && policyConf.Manifests[manifestGroupIndex].Name != "" {
		// If the user does not added Manifests[manifestGroupIndex].Name
		// then return policyConf.Name as a configurationPolicyName
		configName = policyConf.Manifests[manifestGroupIndex].Name
		if manifestIndex > 0 && configName != "" {
			return fmt.Sprintf("%s%d", configName, manifestIndex+1)
		}
	}

	if configName == "" {
		configName = policyConf.Name
		if policyNum > 1 {
			return fmt.Sprintf("%s%d", configName, policyNum)
		}
	}

	return configName
}
```

File: internal/utils.go
Comment (dhaiducek) [reply internal/utils.go]: The ConfigurationPolicy name is customizable in the generator manifest--it won't be set in the manifest in this scenario.

File: internal/utils.go
Comment (dhaiducek) [reply internal/utils.go]: I was thinking the loop would have its own counter that only increments if the name wasn't overridden. So, copying/pasting my other sample, using something like a `templateCounter`:
```
if j != 0 && policyConf.name != policyConf.Manifests[i].Name {
	policyConf.Manifests[i].Name = fmt.Sprintf("%s%d", policyConf.Manifests[i].Name, j+1)
} else {
  templateCounter++
}
```

Comment (yiraeChristineKim) [reply internal/utils.go]: Why templateCounter increase when j == 0?

Comment (dhaiducek) [reply internal/utils.go]: Oh, good point. So it'd be something like this then?
```golang
if j != 0 && policyConf.name != policyConf.Manifests[i].Name {
	policyConf.Manifests[i].Name = fmt.Sprintf("%s%d", policyConf.Manifests[i].Name, j+1)
} else if policyConf.name == policyConf.Manifests[i].Name {
  templateCounter++
}

```

File: internal/utils.go
Comment (yiraeChristineKim) [reply internal/utils.go]: @Dale I added a unit test for `object-templates-raw` When you are free, When you have a moment, could you review it to ensure we're on the same page?

Comment (dhaiducek) [reply internal/utils.go]: Looks good, thanks! I dug into this. It looks like tacking on the index of the policy-template is already the current behavior (for example, providing a ConfigurationPolicy directly would bring its name similar to how this feature does) and adding a counter like I'm proposing would be too different.

File: internal/utils.go
Comment (dhaiducek) [inline internal/utils.go:258]: nit: extra newline
```suggestion
```

File: internal/utils.go
Comment (dhaiducek) [inline internal/utils.go:350]: (optional) `configName` could be assigned to `policyConf.Manifests[manifestGroupIndex].Name` up front. The flow might be easier to understand that way.
```suggestion
	// Do not append a number to configName when it is used for the first time.
	configName := policyConf.Manifests[manifestGroupIndex].Name

	if useManifestName && configName != "" {
		if manifestIndex > 0 {
			return fmt.Sprintf("%s%d", configName, manifestIndex+1)
		}

		return configName
	}

	// Fall back to policy name
	configName = policyConf.Name
	if policyNum > 1 {
		return fmt.Sprintf("%s%d", configName, policyNum)
	}

	return configName
}
```

File: internal/utils.go
Comment (dhaiducek) [inline internal/utils.go:345]: (optional) With the helper function, we could move the increment logic here (and it'd match the manifest code above for consistency).
```suggestion
		if policyNum > 0 {
			return fmt.Sprintf("%s%d", configName, policyNum+1)
```

**Review by dhaiducek — APPROVED** [review]
/hold

I have a bunch of optional comments, so I've approved with a hold if you want to address them but overall looks good to me!

File: internal/utils.go
Comment (yiraeChristineKim) [reply internal/utils.go]: good!

**Review by dhaiducek — APPROVED** [review]

**Pull request merged**
